engine/src/main/java/org/terasology/rendering/openvrprovider/OpenVRProvider.java

```diff
@@ -63,15 +63,37 @@
     //keyboard
     private static boolean keyboardShowing;
     private static boolean headIsTracking;
+    private static OpenVRProvider instance;
 
-    public OpenVRState vrState = new OpenVRState();
+    private static final OpenVRState vrState = new OpenVRState();
 
     // TextureIDs of framebuffers for each eye
     private final VRTextureBounds_t texBounds = new VRTextureBounds_t();
     private float nearClip = 0.5f;
     private float farClip = 500.0f;
 
-    public OpenVRProvider() {
+    // Call getInstance() to instantiate
+    private OpenVRProvider() {
+    }
+
+    // Get a singleton instance.
+    /*
+    * As a general rule, we should use this class as a singleton, because multiple instantiation
+    * will likely cause problems in the upstream native library. This provides a convenient method
+    * of using OpenVRProvider as a singleton.
+     */
```

Comment:
Can you cleanup this javadoc? The comment above it should be integrated in the javadoc and the javadoc should be in the form:

```java
/**
 * profoundly meaningful and masterfully written documentation
 */
```
Notice the double asterisks.